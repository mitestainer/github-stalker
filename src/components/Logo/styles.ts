import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  text-align: center;
`

export const Emoji = styled.span`
  font-size: 12rem;
`

export const Title = styled.h1`
  ${({ theme }) => css`
    font-family: 'Indie Flower', cursive;
    font-size: ${theme.font.size.title};
    background: -webkit-linear-gradient(
      90deg,
      rgba(95, 73, 202, 1) 0%,
      rgba(147, 135, 225, 1) 85%,
      rgba(173, 165, 225, 1) 100%
    );
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
  `}
`
