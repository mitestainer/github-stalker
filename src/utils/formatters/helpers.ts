export const formatDate = (dateString: string) => {
  return new Intl.DateTimeFormat('en-US', {
    day: 'numeric',
    month: 'short',
    year: 'numeric'
  }).format(new Date(dateString))
}
