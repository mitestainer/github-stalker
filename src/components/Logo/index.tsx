import * as S from './styles'

const Logo = () => {
  return (
    <S.Wrapper>
      <S.Emoji aria-label="detective emoji">🕵️</S.Emoji>
      <S.Title>Github stalker</S.Title>
    </S.Wrapper>
  )
}

export default Logo
