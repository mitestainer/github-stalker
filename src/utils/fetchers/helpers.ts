import { formatDate } from 'utils/formatters/helpers'

type RepoResponseProps = {
  id: number
  full_name: string
  description: string
  html_url: string
  watchers: number
  language: string
  updated_at: string
}

export const fetchRepos = async (handle: string, page: number) => {
  const response = await fetch(
    `https://api.github.com/orgs/${handle}/repos?sort=updated&direction=desc&page=${page}&per_page=12`
  )
  const data = await response.json()

  return data.map((repo: RepoResponseProps) => ({
    id: repo.id,
    name: repo.full_name,
    description: repo.description,
    url: repo.html_url,
    stars: repo.watchers,
    language: repo.language,
    updatedAt: formatDate(repo.updated_at)
  }))
}
