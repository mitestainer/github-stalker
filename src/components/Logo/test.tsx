import { render, screen } from 'utils/tests/helpers'

import Logo from '.'

describe('<Logo />', () => {
  it('should render the Logo correctly', () => {
    const { container } = render(<Logo />)

    expect(screen.getByLabelText('detective emoji')).toBeInTheDocument()

    const title = screen.getByRole('heading', { name: /github stalker/i })
    expect(title).toBeInTheDocument()
    expect(title).toHaveStyleRule('font-family', /indie flower/i)

    expect(container.firstChild).toMatchSnapshot()
  })
})
