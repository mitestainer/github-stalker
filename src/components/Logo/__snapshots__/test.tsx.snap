// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render the Logo correctly 1`] = `
.c0 {
  text-align: center;
}

.c1 {
  font-size: 12rem;
}

.c2 {
  font-family: 'Indie Flower',cursive;
  font-size: 8rem;
  background: -webkit-linear-gradient( 90deg,rgba(95,73,202,1) 0%,rgba(147,135,225,1) 85%,rgba(173,165,225,1) 100% );
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

<div
  class="c0"
>
  <span
    aria-label="detective emoji"
    class="c1"
  >
    üïµÔ∏è
  </span>
  <h1
    class="c2"
  >
    Github stalker
  </h1>
</div>
`
